include/picongpu/plugins/common/openPMDWriteMeta.hpp

```diff
@@ -44,24 +44,20 @@ namespace picongpu
 
         namespace writeMeta
         {
-            /** write openPMD species meta data
+            /** write meta data for species
              *
-             * @tparam numSpecies count of defined species
+             * @param threadParams context of the openPMD plugin
+             * @param fullMeshesPath path to mesh entry
              */
-            template<uint32_t numSpecies = bmpl::size<VectorAllSpecies>::type::value>
-            struct OfAllSpecies
+            void ofAllSpecies(::openPMD::Series& series, uint32_t currentStep)
```

Comment:
The function must be `inline`, we aim to use multiple translation units in the future and try to be prepared for this moment and try to avoid ambigious linker issues.

Reply:
Good catch!